apps/api/src/app/environments-v1/usecases/construct-framework-workflow/construct-framework-workflow.usecase.ts

```diff
@@ -103,65 +108,68 @@ export class ConstructFrameworkWorkflow {
             return this.inAppOutputRendererUseCase.execute({ controlValues, fullPayloadForRender });
           },
           // Step options
-          this.constructChannelStepOptions(staticStep)
+          this.constructChannelStepOptions(staticStep, fullPayloadForRender)
         );
       case StepTypeEnum.EMAIL:
         return step.email(
           stepId,
           async (controlValues) => {
             return this.emailOutputRendererUseCase.execute({ controlValues, fullPayloadForRender });
           },
-          this.constructChannelStepOptions(staticStep)
+          this.constructChannelStepOptions(staticStep, fullPayloadForRender)
         );
       case StepTypeEnum.SMS:
         return step.inApp(
           stepId,
           async (controlValues) => {
             return this.smsOutputRendererUseCase.execute({ controlValues, fullPayloadForRender });
           },
-          this.constructChannelStepOptions(staticStep)
+          this.constructChannelStepOptions(staticStep, fullPayloadForRender)
         );
       case StepTypeEnum.CHAT:
         return step.inApp(
           stepId,
           async (controlValues) => {
             return this.chatOutputRendererUseCase.execute({ controlValues, fullPayloadForRender });
           },
-          this.constructChannelStepOptions(staticStep)
+          this.constructChannelStepOptions(staticStep, fullPayloadForRender)
         );
       case StepTypeEnum.PUSH:
         return step.inApp(
           stepId,
           async (controlValues) => {
             return this.pushOutputRendererUseCase.execute({ controlValues, fullPayloadForRender });
           },
-          this.constructChannelStepOptions(staticStep)
+          this.constructChannelStepOptions(staticStep, fullPayloadForRender)
         );
       case StepTypeEnum.DIGEST:
         return step.digest(
           stepId,
           async (controlValues) => {
             return this.digestOutputRendererUseCase.execute({ controlValues, fullPayloadForRender });
           },
-          this.constructActionStepOptions(staticStep)
+          this.constructActionStepOptions(staticStep, fullPayloadForRender)
         );
       case StepTypeEnum.DELAY:
         return step.delay(
           stepId,
           async (controlValues) => {
             return this.delayOutputRendererUseCase.execute({ controlValues, fullPayloadForRender });
           },
-          this.constructActionStepOptions(staticStep)
+          this.constructActionStepOptions(staticStep, fullPayloadForRender)
```

Comment:
Nit-picky: This function is the same across all steps, we can remove it out side the switch.

Reply:
It is not the same one, we have one for Action step and one for Channel step, let's keep it like this for one.